src/garage/tf/policies/categorical_gru_policy2.py

```diff
@@ -67,7 +67,7 @@ def __init__(self,
                  hidden_b_init=tf.zeros_initializer(),
                  recurrent_nonlinearity=tf.nn.sigmoid,
                  recurrent_w_init=tf.initializers.glorot_uniform(),
-                 output_nonlinearity=tf.nn.softmax,
+                 output_nonlinearity=None,
```

Comment:
Is this because it's already done by a submodel? If so, does it ever makes sense to set this parameter? If not, should we delete it?

Reply:
Submodel already applied a softmax layer. This output_nonlinearity allows users to do whatever they want.
Currently all primitives are sharing the same protocol, i.e. hidden_w, hidden_b, hidden_nonlinearity, output_w, output_b, output_nonlinearity.
We can also decide to remove some of them if we think some primitives do not need them, but I think that's not what we should do at this moment.

Reply:
Okay, sounds good.